server/src/test/java/org/elasticsearch/index/engine/LiveVersionMapTests.java

```diff
@@ -396,6 +397,32 @@ public void testPruneTombstonesWhileLocked() throws InterruptedException, IOExce
         assertEquals(0, map.getAllTombstones().size());
     }
 
+    public void testNeverLeaveStaleDeleteTombstone() throws Exception {
```

Comment:
Maybe name the test: "testRandomlyDeletingAndIndexingDoc"? it tests much more, which is good.